### ISSUE TITLE
Question about the training trick

### ISSUE THREAD

**Issue by xingjianleng** [wangguanan/DenoiseRep#4]
Congratulations to the authors for the paper accepted as Neurips'24 Oral.

I have some questions related to the implementation, specifically on the so-called `a training trick to obtain better performance.` https://github.com/wangguanan/DenoiseRep/blob/2facb8e0f655e6c1a0848c527c2d6074bbce8869/Classification/cifar-10/vision-transformers-cifar10/train_cifar10_denoise.py#L252-L258

To my understanding, any `epoch` non-divisible by 10, will enter the outer if block. Then the `epoch /10` will be a floating number and non-divisible by 2. Essentially, other than epoch 0, it will continuously call the `freeze_denoise_layers` function https://github.com/wangguanan/DenoiseRep/blob/2facb8e0f655e6c1a0848c527c2d6074bbce8869/denoiserep_op/denoiserep/denoise_layer.py#L145-L150

So, in essence, what is happening here is
1. For epoch 0, the whole model as well as the denoising layers are trained.
2. For any other later epochs, the original model is continuously trained. However all denoising layers are frozen, and it holds almost the same parameters as from the pretrained model.

Therefore, based on my understanding to the current implementation shown in the repo, it's training the model for more epochs for performance benefits. Please correct me if I'm wrong.

Also, as a follow-up question, could the authors please share the hyper-parameters used in each experiment?

Thanks.

**Comment by muhuagithub** [comment]
Python can perform remainder operations on floating-point numbers. If it is 0.0, it is false, and if it is 1.0, it is true. But personally, I believe that the improvement in model accuracy comes from retraining additional parameters

**Comment by RG12138** [comment]
Thank you for your interest in our work.

Regarding your question, we would like to clarify that the related condition was caused by a bug in the code at the time of submitting the PR, which has been fixed in the subsequent version.

Moreover, we conducted corresponding fairness experiments. Specifically, we trained the baseline model for the same additional number of epochs as our method. The results show that the baseline performance does not further improve with more training. Therefore, the performance gain of our approach mainly comes from the effectiveness of the DenoiseRep module in removing noise, rather than from longer training.

Thank you for your valuable feedback, and we are happy to discuss further if you have any additional questions.
